From: https://github.com/ceph/ceph/pull/63795

From 8886bdd95387bba8c9fcfe0e2550c5eb430d2a69 Mon Sep 17 00:00:00 2001
From: Kefu Chai <tchaikov@gmail.com>
Date: Fri, 13 Jun 2025 16:20:10 +0800
Subject: [PATCH 1/2] cmake: drop c-ares::c-ares alias

Remove the c-ares::c-ares alias that was causing build failures after
bumping the minimum CMake version:

```
CMake Error at cmake/modules/Findc-ares.cmake:34 (add_library):
  add_library cannot create ALIAS target "c-ares::c-ares" because another
  target with the same name already exists.
Call Stack (most recent call first):
  src/CMakeLists.txt:463 (_find_package)
  src/seastar/cmake/SeastarDependencies.cmake:136 (find_package)
  src/seastar/CMakeLists.txt:395 (seastar_find_dependencies)`
```

The alias was originally added for backward compatibility with Seastar,
but is no longer needed since the updated Seastar submodule no longer
references the c-ares::c-ares target.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>
---
 cmake/modules/Findc-ares.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cmake/modules/Findc-ares.cmake b/cmake/modules/Findc-ares.cmake
index 93554ed281460..0e4fcd580127f 100644
--- a/cmake/modules/Findc-ares.cmake
+++ b/cmake/modules/Findc-ares.cmake
@@ -30,9 +30,6 @@ if(c-ares_FOUND)
       IMPORTED_LOCATION "${c-ares_LIBRARY}")
   endif()
 
-  # to be compatible with old Seastar
-  add_library(c-ares::c-ares ALIAS c-ares::cares)
-
   if(NOT TARGET c-ares::c-ares)
     add_library(c-ares::c-ares ALIAS c-ares::cares)
   endif()

From 469d82a1f387f342797636fdfff84f570b22f928 Mon Sep 17 00:00:00 2001
From: Kefu Chai <tchaikov@gmail.com>
Date: Sun, 8 Jun 2025 16:24:51 +0800
Subject: [PATCH 2/2] cmake: bump minimum required CMake version to 3.22.1

Bump the minimum required CMake version from 3.5 to 3.22.1 in both
Ceph and embedded dmclock to address deprecation warnings.

CMake 4.0.2 generates deprecation warnings for projects requiring
versions below 3.10:

```
CMake Deprecation Warning at src/dmclock/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.
```

The upstream CI workflow uses Ubuntu 22.04 (CMake 3.22.1) and CentOS 9
(CMake 3.26.5), so bumping to 3.22.1 maintains compatibility with our
supported build environments while enabling access to newer CMake
features.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>
---
 CMakeLists.txt             | 2 +-
 src/dmclock/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb1a9f07212e9..fde3fed781a37 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.22.1)
 
 project(ceph
   VERSION 20.0.0
diff --git a/src/dmclock/CMakeLists.txt b/src/dmclock/CMakeLists.txt
index 049ea5e312945..52742e846f8af 100644
--- a/src/dmclock/CMakeLists.txt
+++ b/src/dmclock/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5.1)
+cmake_minimum_required(VERSION 3.22.1)
 
 project(dmclock CXX)
 
